src/atsamd/Makefile

```diff
@@ -31,10 +32,9 @@ src-$(CONFIG_HAVE_GPIO_I2C) += atsamd/i2c.c
 src-$(CONFIG_HAVE_GPIO_SPI) += atsamd/spi.c
 src-$(CONFIG_HAVE_SERCOM) += atsamd/sercom.c
 src-$(CONFIG_HAVE_GPIO_HARD_PWM) += atsamd/hard_pwm.c
-src-$(CONFIG_MACH_SAMD21) += atsamd/watchdog.c
-src-$(CONFIG_MACH_SAMD21) += atsamd/clock.c atsamd/timer.c generic/timer_irq.c
-src-$(CONFIG_MACH_SAMX5) += atsamd/samd51_watchdog.c
-src-$(CONFIG_MACH_SAMX5) += atsamd/samd51_clock.c generic/armcm_timer.c
+src-$(CONFIG_MACH_SAMC21) += atsamd/watchdog.c atsamd/samc21_clock.c atsamd/timer.c generic/timer_irq.c
+src-$(CONFIG_MACH_SAMD21) += atsamd/watchdog.c atsamd/clock.c atsamd/timer.c generic/timer_irq.c
+src-$(CONFIG_MACH_SAMX5) += atsamd/watchdog.c atsamd/samd51_clock.c generic/armcm_timer.c
```

Comment:
I'm not sure it is a good idea to merge watchdog.c and samd51_watchdog.c together.  Can samc21 just reuse samd51_watchdog.c?  (It's not necessary to rename the file - on many platforms the prefix is meant to be "an example of" not "exclusive to".)

Reply:
It needs a few different things than samd51, but I don’t really understand the point of separating this file. I can pretty much put it wherever, honestly.